library/spdm_common_lib/libspdm_com_opaque_data.c

```diff
@@ -185,25 +215,28 @@ libspdm_process_opaque_data_supported_version_data(libspdm_context_t *spdm_conte
     *opaque_element_support_version;
     spdm_version_number_t *versions_list;
     spdm_version_number_t common_version;
+    uint8_t version_count;
     bool result;
 
     if (spdm_context->local_context.secured_message_version
         .spdm_version_count == 0) {
         return LIBSPDM_STATUS_SUCCESS;
     }
 
-    if (data_in_size !=
-        libspdm_get_opaque_data_supported_version_data_size(spdm_context)) {
-        return LIBSPDM_STATUS_INVALID_MSG_SIZE;
-    }
     if (libspdm_get_connection_version (spdm_context) >= SPDM_MESSAGE_VERSION_12) {
+        if (data_in_size < sizeof(spdm_general_opaque_data_table_header_t) + sizeof(secured_message_opaque_element_table_header_t) + sizeof(secured_message_opaque_element_supported_version_t)) {
+            return LIBSPDM_STATUS_INVALID_MSG_FIELD;
```

Comment:
I think you may use libspdm_get_untrusted_opaque_data_supported_version_data_size(spdm_context, 1) here, because 1 is minimal.

Reply:
Yeah, Make sense, good point!